### DISCUSSION TITLE
Different ways to make `best_case_worst_case()` (or any function) work with different indicators

### DISCUSSION THREAD

**Discussion by maurolepore** [2DegreesInvesting/tiltIndicatorAfter#245] [Ideas]
* Relates to #242

cc' @kalashsinghal 

--

#243 focuses on one indicator other will come. When the time comes, here are some alternatives ways to implement a `best_case_worst_case()` that works for multiple indicators.

1. Make `best_case_worst_case()` a functional. This implements a polymorphic behaviour in a functional style, where the indicator-specific behaviour is passed as a funciton. It would look somethig like this:

```r
   tilt_profile |>
     best_case_worst_case(.f = case_emissions)

   tilt_profile |>
     best_case_worst_case(.f = case_sector)
```     

2. Make `best_case_worst_case()` a generic. This implements polymorphic behaviour in an object-oriended style, where the indicator-specific behaviour is implemented in specific methods (e.g. best_case_worst_case.tilt_profile_emissions(), where the "tilt_profile_emissions" is a subclass of "tilt_profile". It would look something like this:

```r
   tilt_profile_emissions |>
     best_case_worst_case()

   tilt_profile_sector |>
     best_case_worst_case()
```     

3. Make `best_case_worst_case()` general (indicator-independent). The details of each indicator (column names) may be pass as arguments (3.a.) or "removed" via an adaptor (3.b.). It would look something like this:

```r
  # 3.a.
  tilt_profile |>
    best_case_worst_case(
      grouped_by = "benchmark",
      risk_category = "emissions_profile",
      product_name = "ep_product"
    )

  tilt_profile |>
    best_case_worst_case(
      grouped_by = "benchmark",
      risk_category = "sector_profile",
      product_name = "ep_product"
    )


  # OR
  # 3.b.
  tilt_profile |>
    standardize_column_names() |>
    best_case_worst_case()
```

**Comment by kalashsinghal** [comment]
@maurolepore I used the functional way, but I am interested to apply the generic option in future. :)
